### PR TITLE
Update trademark guidelines in CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by PeterShaggyNoble**
In light of #5599, #4804 and a few others, this PR updates our contributing guidelines on when to include and when to omit a (registered) trademark symbol by adding a line that requires it to be included if the brand itself includes it wherever they use their logo, even at icon sizes.

It also updates the example icon for logos that incorporate the symbol into the design of the logo, as Intel no longer does. Chupa Chups was the first one that sprang to mind for me but probably not the best choice as a showcase icon for our documentation - if anyone can think of a better example let me know.

### CODE REVIEW

**Review by ericcornelissen — APPROVED** [review]
I think this is a good clarification, thanks @PeterShaggyNoble :+1: I'll leave it for at least one more @simple-icons/maintainers to review this before we merge it.

As for the example, I think it's fine like this and would probably fine with any other as well :slightly_smiling_face:

**Review by service-paradis — APPROVED** [review]
This is a good clarification! Thank you @PeterShaggyNoble :)

**Pull request merged**
